### PR TITLE
DOC Better title for example showcasing two different axis on the sam…

### PR DESCRIPTION
**Pull request by NelleV**
Just some documentation updates on plot titles and description.

### CODE REVIEW

**Comment by choldgraf** [comment]
I just added some language that (I think) clarifies @timhoffm 's concerns around "parasite" axes, and also fixes the Circle bug. @NelleV if you're OK with these changes, I'll merge if Circle becomes happy since they've already been approved by tim!

**Comment by NelleV** [comment]
Thanks @choldgraf for fixing the problems :D

File: examples/subplots_axes_and_figures/plot_parasite_axes.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"""
+=====================================================
+Plotting datasets in different units on the same axes
+=====================================================
+
+This example showcases how to plot two different related datasets with
+different units on the same axes. These are often called "parasite" axes,
+as the second axis is an extension of the first.
```
Comment (timhoffm) [inline examples/subplots_axes_and_figures/plot_parasite_axes.py]: Actually, the terminology is not quite consistent. The title says "same axes" which is ok to attract attention and is what the user wants to do semantically.

Proposal:
This example showcases how to plot two different related datasets with
different y-scales on top of each other. To achieve this, we create a second 
Axes that shares the x-axis with the first Axes using `~Axes.twinx()`.
Such Axes are often called "parasite" axes.
+as the second axis is an extension of the first.

Comment (choldgraf) [reply examples/subplots_axes_and_figures/plot_parasite_axes.py]: I like it! just pushed your changes to nelle's branch

**Comment by ImportanceOfBeingErnest** [comment]
The example you are trying to remove here is part of the `axis_grid1` examples. It uses the `axes_grid1.host_subplot` mechanism to create a twin axes. 

You are replacing this with a totally unrelated `twinx` case from the main matplotlib API. 

This seems pretty unreasonable to me. 

If you fear that this might attract too many people into using the axes_grid1 toolkit instead of the main API, the solution would be to add a link on top of the example stating this. I have previously done so e.g. in https://matplotlib.org/devdocs/gallery/axisartist/demo_parasite_axes2.html#sphx-glr-gallery-axisartist-demo-parasite-axes2-py

**Comment by ImportanceOfBeingErnest** [comment]
As for the example you are trying to add here, there is already the corresponding one present in https://matplotlib.org/devdocs/gallery/subplots_axes_and_figures/fahrenheit_celsius_scales.html

which shows how to use twin axes for such use case. It seems useless to add another example. Instead one would link to this existing one and possibly improve its explanatory part. 

In general, the documentation would strongly benefit from people not randomly and blindly moving examples around and having a little bit of a broader view on what exists already.

**Comment by NelleV** [comment]
thanks for the review. I'll close this PR.

**Comment by choldgraf** [comment]
my 2 cents here:

1. I think @ImportanceOfBeingErnest has a good point that we don't necessarily need to *remove* an example from another part of the docs if it was demonstrating something totally different.
2. That said, this example is a more clear, concise demonstration of the `twinx` than what is in the docs. For example, the link you shared exists under the "AxisArtist" subsection. If I were a new user looking to create a parasite axis, I'd avoid this section entirely because it looks like more of an advanced use case.
3. And either way, IMO a new example that has good content, is well-structured, and well-scopred, especially if it's in a different part of the docs, is always an improvement.
4. I'll also gently suggest that comments like:
  
   > In general, the documentation would strongly benefit from people not randomly and blindly moving examples around and having a little bit of a broader view on what exists already.

   are unproductive in pull requests. That said, your choice of language is your own.


that said, it looks like this PR is closed now  :-/

**Comment by NelleV** [comment]
oh, I can still reopen it if other core developers think it's useful. Having not been so active recently in the community, I haven't been following so much what's going on, although I should trust the judgement of the two core developers that have been by far the most active on documentation, and have awarded ~and cited~ research work on the practices surrounding documentation :p

**Comment by NelleV** [comment]
Just reopened. Feel free to edit as much as you want my PR @choldgraf 

In general, we have in the past couple of years tried to *reduce* the number of examples in our gallery, which is ridiculously large (and also largely unmaintained), so if there is an example that I missed that does the same thing, I am more than in favor of just removing it, and I apologize for having wasted y'all's time. 

On the other hand, we should not be documented complicated solution, even in another mpl_toolkit, if a simple one exists.

**Comment by choldgraf** [comment]
I'd vote for:

1. Re-instate the axes_grid1 example, because as @ImportanceOfBeingErnest mentions this was demonstrating a different kind of functionality.
2. Keep this example, and link out to the other two in the docs as more advanced use cases.

I agree with you both that signal-to-noise is a big problem in the documentation. I think there's a time and a place to cull examples (I'd be more than happy to review a PR from somebody that wants to cut down on the number of examples). But, to me the most important question is "do the examples have good content?" I think this one does.

**Comment by NelleV** [comment]
@choldgraf sounds good. I'll remove the last two commits and force push.

**Comment by NelleV** [comment]
(which is something even with core commit rights, I don't believe you can do on my branch).

**Comment by NelleV** [comment]
I'm not sure whether the CI's need to be retriggered manually. I don't often force push to remove commits--this has the unfortunate effect that the commit hash hasn't changed (unlike a rebase), so I have no clue what happens with the CI.

I think I have the rights to retrigger them if need be.

**Comment by ImportanceOfBeingErnest** [comment]
@choldgraf I linked to an [existing example](https://matplotlib.org/devdocs/gallery/subplots_axes_and_figures/fahrenheit_celsius_scales.html) that does already do what this new example would be doing as well. Or maybe you want to comment on where you think they differ? Adding an explanatory text is of great benefit in any case.

In my opinion (sure, who else could I speak for?)
* duplicate examples are to be avoided, noise in the gallery is indeed too high.
* first adding examples and later removing them again is not a useful strategy
* examples showing similar results by different methods should be linked between each other.
* all the toolkits examples could benefit from being critically reviewed and checked for possible simpler solutions inside the main API to link to, but should not be removed without discussion.

The sentence 

>In general, the documentation would strongly benefit from people not randomly and blindly moving examples around and having a little bit of a broader view on what exists already.

can hardly be wrong, can it? And I think the point I wanted to make came across.

**Comment by ImportanceOfBeingErnest** [comment]
@NelleV In general `push --force` will retrigger the testing, given that there is indeed any difference to the last commit.

**Comment by choldgraf** [comment]
Thanks for the comments @ImportanceOfBeingErnest, I'll respond inline below

> linked to an existing example that does already do what this new example would be doing as well

The [existing example](https://matplotlib.org/devdocs/gallery/subplots_axes_and_figures/fahrenheit_celsius_scales.html) does indeed show the use of `twinx`. It also defines two functions, uses a "callback". That makes it more complex than the example originally in this PR, which was short and sweet, and showed the method of interest (`twinx`) early on. I think that makes it a valuable addition.

> duplicate examples are to be avoided, noise in the gallery is indeed too high.

In general I agree with this.

> first adding examples and later removing them again is not a useful strategy

I don't think it's a strategy we should aim for, but one that may nonetheless need to be employed over time. Documentation as complex at Matplotlib's is naturally going to grow organically over time, which means it'll grow semi-chaotically. Occasionally cleaning things up, consolidating example, removing redundancies, etc is a good thing. It's what @NelleV and I did a year ago when we refactored the documentation and I think the docs are better off for it.

> examples showing similar results by different methods should be linked between each other.

yep, definitely agree

> all the toolkits examples could benefit from being critically reviewed and checked for possible simpler solutions inside the main API to link to, but should not be removed without discussion.

also **definitely** agree. though, my 2 cents re: discussion is that documentation in general should be a bit more fluid than the codebase itself. We don't want to create broken links in other websites etc, but in my experience working on documentation is particularly susceptible to [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding). 

> can hardly be wrong, can it? And I think the point I wanted to make came across.

Oftentimes it is less important to be technically correct, and more important to be socially gracious. This is especially true in open source projects where we depend entirely on building a positive, inclusive, and appreciative community. This means that, above all else, we treat people with respect and support. On the contrary when people do **not** do this, it may create short-term gains in terms of making the "right" decision in a code base, but has a stronger long-term drawback in creating a culture that isn't productive and welcoming. I know that you are trying to do what is best for the project, but it's easy to unintentionally come across as condescending, dismissive, or mean, especially on the internet. We need to work extra carefully not to do so.

For reference, I'd suggest reading and internalizing the [Python Software Foundation Code of Conduct](https://www.python.org/psf/codeofconduct/) which Matplotlib also refers to in the ["need help"](https://matplotlib.org/#need-help) section of its main landing page.

**Comment by afvincent** [comment]
Well I may be tired, but @ImportanceOfBeingErnest's [example](https://matplotlib.org/devdocs/gallery/subplots_axes_and_figures/fahrenheit_celsius_scales.html) is showing how to use a twin axis to display a secondary (y-)axis on the very same data (with some kind of trick to auto-update the second one if one plays with the first one IIAC), while @NelleV's [one](https://matplotlib.org/devdocs/gallery/axes_grid1/parasite_simple.html) shows how to display *two* different data sets with the same x-axis but two different y-scales/dimensions. The usage looks quite different to me :/.

But maybe @ImportanceOfBeingErnest was actually looking for [this example](https://matplotlib.org/devdocs/gallery/ticks_and_spines/multiple_yaxis_with_spines.html), which is slightly more evolved than @NelleV's one but indeed close to it. The good news is that this one refers to the others similar examples ;).

**Comment by NelleV** [comment]
@ImportanceOfBeingErnest I'm not so sure that the builds were retriggered. I was looking at the PR quite closely when I forced push, and didn't see the travis-ci trigger. There's always the possibility that the travis-CI did not trigger because it had run already on this commit hash, and it is being smart about this.

**Comment by ImportanceOfBeingErnest** [comment]
One may always opt for first showing an easy, straight forward example and further down on the page showing how to make this code complete, e.g. in the case of the [fahrenheit_celsius_scales](https://matplotlib.org/devdocs/gallery/subplots_axes_and_figures/fahrenheit_celsius_scales.html), first showing how to use `twinx` to create the desired plot, in a second code sniplet adding the callbacks.

I do think that the [fahrenheit_celsius_scales](https://matplotlib.org/devdocs/gallery/subplots_axes_and_figures/fahrenheit_celsius_scales.html) is sufficiently different from the [Multiple Yaxis With Spines](https://matplotlib.org/devdocs/gallery/ticks_and_spines/multiple_yaxis_with_spines.html) example, because that latter one is also mainly concerned with getting the spines to the correct position. 

In any case, as it stands, this updated PR just adds some explanation to the axes_grid1 example, which is of course better than not having this explanation.

**Pull request closed without merging**